src/reputation/views/bounty_view.py

```diff
@@ -426,22 +433,60 @@ def cancel_bounty(self, request, pk=None):
                 # Exception is raised to rollback database transaction
                 raise Exception("Bounty cancel error")
 
+    def filter_queryset(self, queryset):
+        queryset = super().filter_queryset(queryset)
+
+        bounty_types = self.request.query_params.getlist("bounty_type")
+        hub_ids = self.request.query_params.getlist("hub_ids")
+
+        applied_filters = Q()
+
+        # Only return bounties within specific hubs
+        if hub_ids:
+            applied_filters &= Q(unified_document__hubs__id__in=hub_ids)
+
+        # ResearchHub foundation only filter
+        if "researchhub" in bounty_types:
+            researchhub_official_user_accounts = User.objects.filter(
+                is_official_account=True
+            )
+            applied_filters &= Q(created_by__in=researchhub_official_user_accounts)
+
+        # Handle review, answer, and other filters
+        review_or_answer_filter = Q()
+        if "review" in bounty_types:
+            review_or_answer_filter |= Q(bounty_type="REVIEW")
+        if "answer" in bounty_types:
+            review_or_answer_filter |= Q(bounty_type="ANSWER")
+        if "other" in bounty_types:
+            review_or_answer_filter |= Q(bounty_type="OTHER")
```

Comment:
Does a `OTHER` bounty type exist?

Reply:
It does, but it is called "GENERIC_COMMENT" in the database 🫤